Refactor and Using RBOOL macro

S-H-GAMELINKS opened the pull request:
Refactor some function return vaalue and Using `RBOOL` macro.

nobu reviewed (approved):
The indentation in numeric.c is off.

S-H-GAMELINKS commented:
@nobu 
Thanks for your comments.

Removed `RBOOL` macro and using `rb_equal` function return value.
And, fixed indent in `fix_gt`, `fix_ge`, `fix_lt` and `fix_le` in numeric.c
